index.js

```diff
@@ -832,12 +832,12 @@ RingPop.prototype.handleOrProxy =
                 key: key,
                 url: req && req.url
             });
-            this.proxyReq(_.extend(opts, {
+            this.proxyReq(_.extend({
```

Comment:
this is not the semantics you want.

You wanted `_.extend(opts || {}, { ... })`

If you set `{ key: 'foo' }` then other stuff will break horribly downstream.